src/tcache.c

```diff
@@ -187,31 +187,36 @@ tcache_event(tsd_t *tsd) {
 	bool is_small = (szind < SC_NBINS);
 	cache_bin_t *cache_bin = &tcache->bins[szind];
 
-	tcache_bin_flush_stashed(tsd, tcache, cache_bin, szind, is_small);
+	if (!tcache_bin_disabled(szind, cache_bin, tcache_slow)) {
```

Comment:
Let's handle the less common case in the branch, instead of including the main logic inside which adds another indent level.
```
  if (tcache_bin_disabled()) {
    goto label_done;
  }
  ...
label_done:
  tcache_slow->next_gc_bin++;
  ...
```

Reply:
BTW was this the issue you were debugging? If so let's add more bin enabled assertions into the lower level tcache facilities that can help catching these.

Reply:
Not really, the bug I was debugging was purely in the test itself. This logical flaw was found by the assertion in ncached_max_get. So for most cache_bin facilities (I guess you mean cache_bin?) they should be safe. Let me also double check other utility functions there.

Reply:
Also the tcache functions. They should only be invoked only when the corresponding bin is enabled, so adding assertions there would help too. I'll leave it to you to decide whether/where to assert.